test/with_dummyserver/test_socketlevel.py

```diff
@@ -842,3 +842,34 @@ def test_empty_head_response_does_not_hang(self):
 
         # stream will use the read method here.
         self.assertEqual([], list(r.stream()))
+
+
+class TestChunkedTransfer(SocketDummyServerTestCase):
+    def test_chunked(self):
+        self.buffer = b''
+
+        def socket_handler(listener):
+            sock = listener.accept()[0]
+
+            while not self.buffer.endswith('\r\n0\r\n\r\n'):
+                self.buffer += sock.recv(65536)
+
+            # Send incomplete message (note Content-Length)
+            sock.send(
+               b'HTTP/1.1 200 OK\r\n'
+               b'Content-type: text/plain\r\n'
+               b'Content-Length: 0\r\n'
+               b'\r\n')
+            sock.close()
+
+        self._start_server(socket_handler)
+        chunks = ['foo', 'bar', 'bazzzzzz']
+        pool = HTTPConnectionPool(self.host, self.port, retries=False)
+        r = pool.urlopen('GET', '/', chunks, headers=dict(DNT='1'), chunked=True)
+
+        self.assertTrue('Transfer-Encoding' in self.buffer)
+        body = self.buffer.split('\r\n\r\n')[1]
+        lines = body.split('\r\n')
+        for i, chunk in enumerate(chunks):
+            self.assertEqual(lines[i * 2], str(len(chunk)))
+            self.assertEqual(lines[i * 2 + 1], chunk)
```

Comment:
Agreed, so far it only checks the graceful handling, no errors or oddities. I'll implement your suggestion as tests, open for further suggestions.